SolidStateGroup/bullet-train-api
# feat: Issue 166 json formatted logs

Thanks for submitting a PR! Please check the boxes below:


## Changes

Extending logging.Formatter with a custom class for json formatted messages.  Unfortunately specifying json format as a string in the LOGGING dictionary presented a corner-case with the axes module (which returns a dictionary [with double-quoted values] with request information).  Traditional string formatting tokens also had to be replaced with '{}' since the tokens were not being interpolated properly after the logging formatter was added.


## How did you test this code?
1. Build `flagsmith:local` container locally.
2. Update `docker-compose.yaml` to use `flagsmith:local` container.
3. Update environment variables in docker-compose file to include `LOG_FORMAT: json`
4. Basic local testing through the UI to ensure logs were properly formatted.
